app/components/post_textbox/post_textbox_base.js

```diff
@@ -577,22 +586,49 @@ export default class PostTextBoxBase extends PureComponent {
 
         const currentMembersCount = this.props.currentChannelMembersCount;
         const notificationsToChannel = this.props.enableConfirmNotificationsToChannel && this.props.useChannelMentions;
+        const notificationsToGroups = this.props.enableConfirmNotificationsToChannel && this.props.useGroupMentions;
         const toAllOrChannel = this.textContainsAtAllAtChannel(value);
-
+        let groupMentions = this.groupsMentionedInText(value);
         if (value.indexOf('/') === 0) {
             this.sendCommand(value);
         } else if (notificationsToChannel && currentMembersCount > NOTIFY_ALL_MEMBERS && toAllOrChannel) {
             this.showSendToAllOrChannelAlert(currentMembersCount);
+        } else if (notificationsToGroups && !toAllOrChannel && groupMentions.length > 0) {
+            let memberNotifyCount = 0;
+            let channelTimezoneCount = 0;
+
+            groupMentions = groupMentions.
+                map((group) => {
+                    const mappedValue = this.props.channelMemberCountsByGroup[group.id];
+                    if (mappedValue && mappedValue.channel_member_count > NOTIFY_ALL_MEMBERS && mappedValue.channel_member_count > memberNotifyCount) {
+                        memberNotifyCount = mappedValue.channel_member_count;
+                        channelTimezoneCount = mappedValue.channel_member_timezones_count;
+                    }
+                    return `@${group.name}`;
+                });
+            groupMentions = [...new Set(groupMentions)];
+            if (memberNotifyCount > 0) {
+                this.showSendToGroupsAlert(groupMentions, memberNotifyCount, channelTimezoneCount);
+            } else {
+                this.doSubmitMessage();
+            }
         } else {
             this.doSubmitMessage();
         }
     };
 
     textContainsAtAllAtChannel = (text) => {
-        const textWithoutCode = text.replace(/(`+)([^`]|[^`][\s\S]*?[^`])\1(?!`)| *(`{3,}|~{3,})[ .]*(\S+)? *\n([\s\S]*?\s*)\3 *(?:\n+|$)/g, '');
+        const textWithoutCode = text.replace(CODE_REGEX, '');
         return (/\B@(all|channel)\b/i).test(textWithoutCode);
     };
 
+    groupsMentionedInText = (text) => {
+        const {groupsWithAllowReference} = this.props;
+        const textWithoutCode = text.replace(CODE_REGEX, '');
+        const mentions = textWithoutCode.match(AT_MENTION_REGEX_GLOBAL) || [];
+        return (mentions.length > 0 && mentions.map((mention) => groupsWithAllowReference && groupsWithAllowReference.get(mention)).filter((trueVal) => trueVal)) || [];
```

Comment:
Hey @migbot,

I realised that I actually do need to keep the map because mentions is just an array of mentions ( ["@developers", "@qa"]) and I want to replace them with the group information so when I map it I get something like [developersGroupObject, qaGroupObject] which contains data like group ID and group name that I use in the `forEach` loop. I kept the work we did but just chained along a `.map`. Let me know if you think that's okay if there is a better way to do it.